src/Avalonia.Vulkan/IVulkanDevice.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using Avalonia.Metadata;
+using Avalonia.Platform;
+using Avalonia.Vulkan.UnmanagedInterop;
+
+namespace Avalonia.Vulkan;
+public interface IVulkanDevice : IDisposable, IOptionalFeatureProvider
+{
+    public IntPtr Handle { get; }
+    public IntPtr PhysicalDeviceHandle { get; }
+    public IntPtr MainQueueHandle { get; }
+    public uint GraphicsQueueFamilyIndex { get; }
+    public IVulkanInstance Instance { get; }
+    bool IsLost { get; }
+    public IDisposable Lock();
+    public IEnumerable<string> EnabledExtensions { get; }
+}
+
+public interface IVulkanInstance
+{
+    public IntPtr Handle { get; }
+    public IntPtr GetInstanceProcAddress(IntPtr instance, string name);
+    public IntPtr GetDeviceProcAddress(IntPtr device, string name);
+    public IEnumerable<string> EnabledExtensions { get; }
+}
+
+[NotClientImplementable]
+public interface IVulkanPlatformGraphicsContext : IPlatformGraphicsContext
+{
+    IVulkanDevice Device { get; }
+    IVulkanInstance Instance { get; }
+    internal VulkanInstanceApi InstanceApi { get; }
+    internal VulkanDeviceApi DeviceApi { get; }
```

Comment:
Does mix of internal and public interface members work on netstandard? I am surprised it's compiled.

Reply:
I guess it works.